docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,300 @@
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: ["k8s.io/node", "k8s.io/az"]
+```
+
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+![Use case 2](assets/031/use_case_2.png)
+
+#### Prefer traffic in same zone with fallback to any other zone
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        fallback: Any
+```
+
+![Use case 3](assets/031/use_case_3_1.png)
+![Use case 3](assets/031/use_case_3_2.png)
+
+#### Load balance equally to list of zones
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1", "zone-3"]
+        fallback: Fail
+```
+
+![Use case 4](assets/031/use_case_4_1.png)
+![Use case 4](assets/031/use_case_4_2.png)
+![Use case 4](assets/031/use_case_4_3.png)
+
+#### When leaving zone, target specific zones first
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+        - groups:
+          - zones: ["zone-3"]
+```
+
+![Use case 5](assets/031/use_case_5_1.png)
+![Use case 5](assets/031/use_case_5_2.png)
+![Use case 5](assets/031/use_case_5_3.png)
+
+#### Exclude single zone from load balancing
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - groups:
+          - zones: ["zone-1"]
+        fallback: Fail
+```
+
+![Use case 6](assets/031/use_case_6_1.png)
+![Use case 6](assets/031/use_case_6_2.png)
+![Use case 6](assets/031/use_case_6_3.png)
+
+TODO What if i have lots of zones and don’t want to write all of them during the incident. I would like to just exclude single zone, do we want to add something like:
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAwareness:
+      inZone: []
+      crossZone:
+        - exclude:
+            zones: ["zone-3"]
+```
+
+#### Load balance equally to local and zone
+
+![Use case 7](assets/031/use_case_7.png)
+
+We have the couple of use cases for this, but no config yet:
+- You are migrating from universal to k8s, both zones are in the same AZ and I would like to move traffic gradually from one to another
+- You have have some zone with testing or preview environment and you would like to handle live traffic on it for testing. Is it a use case for locality aware lb? Or should this be done with MeshHTTPRoute?
+The question is if we really need it, or can this be done currently in some other way?
+
```

Comment:
We're missing an example. The case where I have multiple sets of zones that I don't want to overlap.

For example I have us-east and us-west and the other side is eu-west and eu-east.

What I want is "us-east/west" failover and "eu-east/west" failover with no risk of having cross ocean traffic.

Originnaly we thought about writing this:

```yaml
crossZones:
  groups:
     - zones: ["us-east", "us-west"]
     - zones: ["eu-east", "eu-west"]
  fallback: Fail
```

but I feel it goes against other examples here.